src/coreclr/vm/excep.cpp

```diff
@@ -3062,14 +3059,26 @@ void GetExceptionForHR(HRESULT hr, OBJECTREF* pProtectedThrowable)
 
     // Get an IErrorInfo if one is available.
     IErrorInfo *pErrInfo = NULL;
-#ifdef FEATURE_COMINTEROP
     if (SafeGetErrorInfo(&pErrInfo) != S_OK)
         pErrInfo = NULL;
-#endif // FEATURE_COMINTEROP
 
     GetExceptionForHR(hr, pErrInfo, pProtectedThrowable);
 }
+#else
+void GetExceptionForHR(HRESULT hr, OBJECTREF* pProtectedThrowable)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;        // be consistent with COM interop enabled
+        MODE_ANY;
```

Comment:
```suggestion
        GC_TRIGGERS;
        NOTHROW;
        MODE_COOPERATIVE;
```

I'm pretty sure the above is what you want. The `CLRException::GetThrowable()` has a nice contract.

Reply:
I'm really confused now.
The COM-enabled version of `GetExceptionForHR(HRESULT hr, OBJECTREF* pProtectedThrowable)` has MODE_ANY, and calls `GetExceptionForHR(HRESULT hr, IErrorInfo* pErrInfo, OBJECTREF* pProtectedThrowable)` which has MODE_COOPERATIVE. How can it succeed? Should it be fixed?

Reply:
`MODE_ANY` is just that, any mode. Which means calling Cooperative or Preemptive is acceptable. What is means though is that the COM-enabled version can also be more explicit about what is requires. In this case `MODE_COOPERATIVE`.

It isn't uncommon for people to get frustrated by the contracts and just slap the most permissive on them. It is also possible the underlying functions have changed. Updating both seems appropriate.

Reply:
If a caller is in preemptive mode, it will pass at `MODE_ANY` and fail at `MODE_COOPERATIVE` check. Since an `OBJECTREF` is touched I think cooperative is correct.